source/_integrations/lmstudio.markdown

```diff
@@ -0,0 +1,271 @@
+
+Controlling Home Assistant is an experimental feature that provides the AI access to the Assist API of Home Assistant. You can control what devices and entities it can access from the {% my voice_assistants title="exposed entities page" %}. The AI is able to provide you information about your devices and control them.
+
+This integration does not integrate with [sentence triggers](/docs/automation/trigger/#sentence-trigger).
+
+## Prerequisites
+
+To use this integration, you need to have an LM Studio server running on your local machine or on a computer on the local network. 
+- LM Studio is a desktop application that lets you run Large Language Models (LLMs) locally, using an API that is compatible with OpenAI. 
+- You can download LM Studio for macOS, Linux, or Windows from the [official website](https://lmstudio.ai/). 
+- After installing LM Studio, open the application and start the local server. Make sure the server is accessible on your network, usually at `http://YOUR_LOCAL_IP:1234`.
+
+{% include integrations/config_flow.md %}
+
+{% include integrations/option_flow.md %}
+
+{% configuration_basic %}
+Base URL:
+  description: The URL of the LM Studio local server, such as `http://localhost:1234` or `http://192.168.1.100:1234`.
+API key:
+  description: Optional API key for authentication if you've configured one in LM Studio. Leave empty if no authentication is required.
+Model:
+  description: Name of the model to use. The integration will automatically fetch available models from your LM Studio server during setup.
+Instructions:
+  description: Instructions for the AI on how it should respond to your requests. It is written using [Home Assistant Templating](/docs/configuration/templating/).
+Control Home Assistant:
+  description: If the model is allowed to interact with Home Assistant. It can only control or provide information about entities that are [exposed](/voice_control/voice_remote_expose_devices/) to it. This feature is considered experimental. For details on model limitations, refer to the section [Controlling Home Assistant](#controlling-home-assistant).
+Max tokens:
+  description: Maximum number of tokens to generate in the response. Higher values allow for longer responses but may increase processing time.
+Temperature:
+  description: Controls randomness in the AI's responses. Lower values (0.1-0.3) make responses more focused and deterministic, while higher values (0.7-1.0) make them more creative and varied.
+Top P:
+  description: Controls diversity via nucleus sampling. Lower values focus on more probable tokens, while higher values allow more diverse responses.
+Max history messages:
+  description: Maximum number of messages to keep for each conversation (0 = no limit). Limiting this value will cause older messages in a conversation to be dropped.
+{% endconfiguration_basic %}
+
+## Configuration options — detailed guide
```

Comment:
why spread the description of the options into 2 sections? I think these 2 configuration_basic descriptions could be merged into 1.